### DISCUSSION TITLE
Sensor_Value helper methods all assume the precision is in micro unit without documenting this.

### DISCUSSION THREAD

**Discussion by kmeurink** [zephyrproject-rtos/zephyr#82964] [Q&A]
I am currently working on creating a driver that provides one of its measured values in mili Celsius.
I wanted to use the helper methods in the sensor file to convert the sensor_value struct provided by this driver to a float.
However it seems these methods assume the precision is in micro units.

This is not described anywhere, so it is quite easy to make mistakes using these functions. I would therefore say it is a bug, can anyone confirm that this is indeed overlooked behaviour?
Or in the case that there is documentation somewhere that requires the precision of measured signal to be on the micro unit level, where would this be located?

**Comment by kartben** [comment]
@kmeurink do you mean, this? https://docs.zephyrproject.org/latest/doxygen/html/structsensor__value.html
val2 is being described as "Fractional part of the value (in one-millionth parts)". Is there some other place specifically where you feel things are not clear or could be improved?

**Reply by kmeurink** [reply]
Ah, I do not know how I missed that part. It seems I looked at all the function descriptions but not the description of the sensor_value struct, at least not in detail. Thank you for pointing it out.

The only improvement could perhaps be to add a note to the "implement sensor driver" documentation, that specifies that the sensor driver gain needs to be in one-millonth parts?
As that was the first place I looked at to see if there were any guidelines that the sensor driver had to follow. So with a note you are reminded of it when you are working on implementing a driver.
